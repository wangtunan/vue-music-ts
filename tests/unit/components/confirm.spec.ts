import { shallowMount, Wrapper } from '@vue/test-utils'
import Confirm from '@/components/confirm/index.vue'

describe('confirm.vue', () => {
  let wrapper: Wrapper<Vue>
  beforeEach(() => {
    wrapper = shallowMount(Confirm)
  })

  it('match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('default props', () => {
    expect(wrapper.props('visible')).toBe(false)
    expect(wrapper.props('message')).toBe('')
    expect(wrapper.props('cancelButtonText')).toBe('取消')
    expect(wrapper.props('confirmButtonText')).toBe('确定')
    expect(wrapper.props('showCancelButton')).toBe(true)
    expect(wrapper.props('showConfirmButton')).toBe(true)
  })
})