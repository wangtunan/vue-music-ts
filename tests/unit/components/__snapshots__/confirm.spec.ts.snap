// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`confirm.vue match snapshot 1`] = `
<transition-stub name="fade">
  <div class="m-confirm" style="display: none;">
    <div class="confirm-box">
      <h2 class="text"></h2>
      <div class="btn-box">
        <div class="btn cancel">取消</div>
        <div class="btn confirm">确定</div>
      </div>
    </div>
  </div>
</transition-stub>
`;
